src/tezos_interop/tezos_js_bridge.js

```diff
@@ -193,6 +201,59 @@ const onStorageRequest = async (id, content) => {
   respond(id, { status: "success", storage });
 };
 
+/** @param {BigMapMultipleKeysRequest} content */
+const onBigMapMultipleKeyRequest = async (id, content) => {
+  const { rpc_node, confirmation, destination, keys } = content;
+  const client = new RpcClient(rpc_node);
+  const Tezos = new TezosToolkit(rpc_node);
+  Tezos.setProvider({ config });
+
+  const block = await client.getBlock(); // fetches the head
+
+  let values = await Tezos.contract
+    .at(destination)
+    .then((contract) => {
+      return contract.storage();
+    })
+    .then((storage) => {
+      return storage.getMultipleValues(
+        keys
+      );
+    })
+    .then((values) => {
+      let result = new Array(keys.length);
+      values.forEach((value, key) => {
+        let index = keys.indexOf(key);
+        if (value !== undefined) {
+          result[index] = value[1];
+        }
+        else {
+          result[index] = null;
+        }
+      });
+      return result;
+    });
```

Comment:
Please use async await properly, also remove the for loop, it's not clear to me the intention, combine key and values?